connectors/filtering/basic_rule.py

```diff
@@ -147,6 +147,50 @@ def from_string(cls, string):
                 )
 
 
+class BasicRuleValidationResult:
+    def __init__(self, rule_id, is_valid, validation_message):
+        self.rule_id = rule_id
+        self.is_valid = is_valid
+        self.validation_message = validation_message
+
+    @classmethod
+    def valid_result(cls, rule_id):
+        return BasicRuleValidationResult(
+            rule_id=rule_id, is_valid=True, validation_message="Valid rule"
+        )
+
+
+class BasicRuleValidator:
+    @classmethod
+    def validate(cls, rule):
+        raise NotImplementedError
+
+
+class BasicRuleNoMatchAllRegexValidator(BasicRuleValidator):
+    MATCH_ALL_REGEXPS = [".*", "(.*)"]
+
+    @classmethod
+    def validate(cls, basic_rule_json):
+        basic_rule = BasicRule.from_json(basic_rule_json)
+
+        # default rule uses match all regex, which is intended
+        if basic_rule.is_default_rule():
+            return BasicRuleValidationResult.valid_result(rule_id=basic_rule.id_)
+
+        if basic_rule.rule == Rule.REGEX and any(
+            match_all_regex == basic_rule.value
+            for match_all_regex in BasicRuleNoMatchAllRegexValidator.MATCH_ALL_REGEXPS
+        ):
+            return BasicRuleValidationResult(
+                rule_id=basic_rule.id_,
+                is_valid=False,
+                validation_message=f"Match all regexps: '{BasicRuleNoMatchAllRegexValidator.MATCH_ALL_REGEXPS}' are "
+                f"not allowed.",
```

Comment:
I am surprised the linter did not complain (`f` not required)

Reply:
I think the autoformatter split it into these two lines and appended the `f`. Looks also strange to me tbh.